### PR TITLE
WidgetStyle Presence.

### PR DESCRIPTION
**Pull request by DiscordAnaxes**
Here some images of the presence working:

### CODE REVIEW

File: websites/W/WidgetStyle/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "service": "Widgetstyle",
+    "author": {
+      "name": "Anaxes",
+      "id": "567885938160697377"
+    },
+    "category": "other",
+    "logo": "https://cdn.tixte.com/uploads/cdn.anaxes.xyz/kofaevuwx9a.png",
+    "thumbnail": "https://cdn.tixte.com/uploads/cdn.anaxes.xyz/kof75u24d9a.png",
```
Comment (SlowLife1661) [inline websites/W/WidgetStyle/dist/metadata.json:10]: Upload to imgur

File: websites/W/WidgetStyle/presence.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
+const presence = new Presence({
+  clientId: "828789217793409054"
+}), browsingStamp = Math.floor(Date.now() / 1000); 
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  };
```
Comment (SlowLife1661) [inline websites/W/WidgetStyle/presence.ts:8]: put `startTimestamp` here and remove others

File: websites/W/WidgetStyle/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "service": "Widgetstyle",
```
Comment (SlowLife1661) [inline websites/W/WidgetStyle/dist/metadata.json:3]: Must be same as folder name

File: websites/W/WidgetStyle/presence.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
+const presence = new Presence({
+  clientId: "828789217793409054"
+}), browsingStamp = Math.floor(Date.now() / 1000); 
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  }; 
+presenceData.startTimestamp = browsingStamp;
```
Comment (SlowLife1661) [inline websites/W/WidgetStyle/presence.ts:9]:
```suggestion
  const presenceData: PresenceData = {
    largeImageKey: "logo",
    startTimestamp: browsingStamp
  }; 
```

**Comment by ririxidev** [comment]
Yeah um, for now - presence is qualified as low quality becuase of very low amount of DOM Readings etc - technically this presence  is static.

**Comment by DiscordAnaxes** [comment]
> Yeah um, for now - presence is qualified as low quality becuase of very low amount of DOM Readings etc - technically this presence is static.

Oh alright.

File: websites/W/WidgetStyle/dist/metadata.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    "logo": "https://i.imgur.com/DGvcIXh.png",
+    "thumbnail": "https://i.imgur.com/HYYkUG6.png",
+    "version": "1.0.0",
+    "color": "#7289da",
+    "tags": [
+      "discord",
+      "widgetstyle",
+      "widget",
+      "widget.style"
+    ],
```
Comment (ririxidev) [inline websites/W/WidgetStyle/dist/metadata.json:18]:
```suggestion
    "tags": [
      "discord",
      "widgetstyle",
      "widget"
    ],
```

File: websites/W/WidgetStyle/presence.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        {
+          label: "🌐 View Page",
+          url: document.location.href
+        }
+      ];
+    } else if (document.location.pathname.includes("/privacy")) {
+      presenceData.details = "Viewing 🎭 privacy policy page";
+      presenceData.buttons = [
+        {
+          label: "🌐 View Page",
```
Comment (ririxidev) [inline websites/W/WidgetStyle/presence.ts:50]:
```suggestion
          label: "View Page",
```

File: websites/W/WidgetStyle/presence.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+          url: document.location.href
+        }
+      ];
+    } else if (document.location.pathname.includes("/terms")) {
+      presenceData.details = "Viewing 📖 tos page";
+      presenceData.buttons = [
+        {
+          label: "🌐 View Page",
```
Comment (ririxidev) [inline websites/W/WidgetStyle/presence.ts:58]:
```suggestion
          label: "View Page",
```

File: websites/W/WidgetStyle/presence.ts [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+        }
+      ];
+    } else if (document.location.pathname.includes("/user/")) {
+      const user = document.getElementById("username"),
+      usertotal = document.getElementById("usertotal")?.getAttribute("usertotal");
+      presenceData.details = `Viewing 👤 user: ${user.innerText}`;
+      presenceData.state = `They have ${usertotal || "0"} widget(s)`;
+      presenceData.buttons = [
+        {
+          label: `🌐 View ${user.innerText}'s Page`,
```
Comment (ririxidev) [inline websites/W/WidgetStyle/presence.ts:69]:
```suggestion
          label: `View ${user.innerText}'s Page`,
```

**Review by ririxidev — CHANGES_REQUESTED** [review]

File: websites/W/WidgetStyle/presence.ts [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+  clientId: "828789217793409054"
+}), browsingStamp = Math.floor(Date.now() / 1000); 
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo",
+    startTimestamp: browsingStamp
+  }; 
+
+    if (document.location.pathname == "/") {
```
Comment (ririxidev) [inline websites/W/WidgetStyle/presence.ts:11]:
```suggestion
    if (document.location.pathname === "/") {
```

**Review by ririxidev — APPROVED** [review]

File: websites/W/WidgetStyle/presence.ts [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+  }; 
+
+    if (document.location.pathname === "/") {
+      presenceData.details = "Viewing 🏠 home page";
+      presenceData.buttons = [
+        {
+          label: "View Page",
+          url: document.location.href
+        }
+      ];
```
Comment (Bas950) [inline websites/W/WidgetStyle/presence.ts:18]: buttons to the homepage are not allowed.

File: websites/W/WidgetStyle/tsconfig.json [hunk 11]
```diff
@@ -0,0 +1,7 @@
+{
+    "extends": "../../../tsconfig.json",
+    "compilerOptions": {
+      "outDir": "./dist/"
+    }
+  }
+
```
Comment (Bas950) [inline websites/W/WidgetStyle/tsconfig.json:7]:
```suggestion
  }
```

File: websites/W/WidgetStyle/dist/metadata.json [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+      "widget"
+    ],
+    "url": "widget.style",
+    "description": {
+      "fr": "Un look moderne sur les widgets Discord obsolètes.",
+      "tr": "Güncelliğini yitirmiş Discord widget'larına modern bir görünüm.",
+      "ja_JP": "古いDiscordウィジェットのモダンな外観。",
+      "zh_CN": "过时的Discord小部件的现代外观。",
+      "en": "A modern look on outdated Discord Widgets."
+    }
```
Comment (Bas950) [inline websites/W/WidgetStyle/dist/metadata.json:25]:
```suggestion
    "description": {
      "nl": "Een moderne blik op verouderde Discord Widgets.",
      "fr": "Un look moderne sur les widgets Discord obsolètes.",
      "tr": "Güncelliğini yitirmiş Discord widget'larına modern bir görünüm.",
      "ja_JP": "古いDiscordウィジェットのモダンな外観。",
      "zh_CN": "过时的Discord小部件的现代外观。",
      "en": "A modern look on outdated Discord Widgets."
    }
```

File: websites/W/WidgetStyle/dist/metadata.json [hunk 13]
```diff
@@ -0,0 +1,26 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "service": "WidgetStyle",
+    "author": {
+      "name": "Anaxes",
+      "id": "567885938160697377"
+    },
+    "category": "other",
+    "logo": "https://i.imgur.com/5nAdPH8.png",
```
Comment (Bas950) [inline websites/W/WidgetStyle/dist/metadata.json:9]: 514x512 xD please just make it square...

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by ririxidev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**